homeassistant/components/sma/config_flow.py

```diff
@@ -160,6 +159,9 @@ async def async_step_discovery_confirm(
             data_schema=vol.Schema(
                 {
                     vol.Required(CONF_HOST, default=self._data[CONF_HOST]): cv.string,
+                    vol.Optional(
+                        CONF_MAC, default=self._discovery_data[CONF_MAC]
+                    ): cv.string,
```

Comment:
We should not ask the user the mac, we don't need it _that_ much. As in, if we know it via discovery, great! Let's add it. If we find a new mac for an already configured serial number, we can add it along the way and give them the same experience